joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/utils/FileUtil.scala

```diff
@@ -117,12 +117,12 @@ object FileUtil {
     def absolutePathAsString: String = p.toAbsolutePath.toString
 
     def /(child: String): Path = {
-      p.resolve(child)
+      p.resolve(child).toAbsolutePath.normalize()
```

Comment:
Why to we need the absolute path conversion and the normalization?
To do this always even if people only want the `resolve` functionality seems odd.

Reply:
This took me a while to track down - in `better.files.File` we have the following:
```scala
def /(child: String): File = path.resolve(child)

implicit def apply(path: Path): File = new File(path.toAbsolutePath.normalize())

def apply(path: String, fragments: String*): File = Paths.get(path, fragments: _*)
```

From my understanding, since the `/` function returns a `File` type, the `implicit apply` is going to be called which in turn calls the `path.toAbsolutePath.normalize()`. This is what lead me to then updating the `/` function to call the `toAbsolutePath.normalize()`.

There were some issues with resolving directories with an included `./` (e.g. `./someDir`) that this ended up fixing which is why I kept it in.

Reply:
```scala
def /(child: String): File = path.resolve(child)
implicit def apply(path: Path): File = new File(path.toAbsolutePath.normalize())
```

`path.resolve(child)` above returns type `java.nio.file.Path`, which then calls the `implicit apply(path)` which returns the `File`

```scala
val foo = File(someDir) // `better.files.File`
val bar = foo / "cpg.bin" // equivalent to calling the `implicit apply(foo.resolve("cpg.bin"))`
```